ToolBox/FxcmDownloader/FxcmDataDownloader.cs

```diff
@@ -163,35 +161,37 @@ public IEnumerable<BaseData> Get(Symbol symbol, Resolution resolution, DateTime
             // initialize session
             RequestTradingSessionStatus();
 
-            Console.WriteLine("Downloading data from {0} to {1}...", startUtc.ToShortDateString(), endUtc.ToShortDateString());
+            Console.WriteLine("Downloading {0} data from {1} to {2}...", resolution, startUtc.ToString("yyyyMMdd HH:mm:ss"), endUtc.ToString("yyyyMMdd HH:mm:ss"));
+
+            //Find best FXCM  paramrs
+            IFXCMTimingInterval interval = ToFXCMInterval(resolution);
 
-            // download bars
-            var totalBars = new List<TradeBar>();
+            var totalTicks = (endUtc - startUtc).Ticks;
 
-            // calculate the maximum time span for one request (using 10-second bars)
-            const int maxBarsPerRequest = 300;
-            var timeSpanPerRequest = TimeSpan.FromSeconds(maxBarsPerRequest * 10);
+            // download data
+            var totalBaseData = new List<BaseData>();
 
-            var start = startUtc;
-            var end = startUtc + timeSpanPerRequest;
+            var end = endUtc;
 
-            // request loop
-            while (start < endUtc.AddDays(1))
+            do // 
             {
-                _currentBars.Clear();
+                //show progress
+                progressBar(Math.Abs((end - endUtc).Ticks), totalTicks, Console.WindowWidth / 2,'█');
```

Comment:
Yes :) I love it. The desktop LEAN will be epic.

On Tue, Dec 22, 2015 at 11:02 AM, Michael <notifications@github.com> wrote:

> In ToolBox/FxcmDownloader/FxcmDataDownloader.cs
> <https://github.com/QuantConnect/Lean/pull/232#discussion_r48267221>:
>
> >              {
> > -                _currentBars.Clear();
> > +                //show progress
> > +                progressBar(Math.Abs((end - endUtc).Ticks), totalTicks, Console.WindowWidth / 2,'█');
>
> Just making sure @jaredbroad <https://github.com/jaredbroad> sees this,
> he'll love this idea!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/QuantConnect/Lean/pull/232/files#r48267221>.
>



-- 
Jared Broad